src/utils/entitySpec.js

```diff
@@ -32,6 +41,67 @@ export default class EntitySpec {
     return new EntitySpec(type, provider, namespace, name, revision, pr)
   }
 
+  static checkDroppedFiles(files) {
+    let acceptedFiles = []
+    let rejectedFiles = []
+
+    files.forEach(file => {
+      const acceptedType = acceptedFilesValues.find(el => el === file.type)
+
+      if (acceptedType) acceptedFiles.push(file)
+      else rejectedFiles.push(file)
+    })
+
+    return {
+      acceptedFiles,
+      rejectedFiles
+    }
+  }
+
+  static fromUrl(url) {
+    const urlObject = new URL(url)
+    const pathname = urlObject.pathname.startsWith('/') ? urlObject.pathname.slice(1) : urlObject.pathname
+    const [packageName, name, version, revision] = pathname.split('/')
+    const hostname = urlObject.hostname.replace('www.', '')
+    let path
+
+    switch (hostname) {
+      case NPM_WEBSITE:
+        if (revision) path = `npm/npmjs/-/${name}/${revision}`
+        else path = `npm/npmjs/-/${name}`
+        return path
+
+      case GITHUB_WEBSITE:
+        if (revision) path = `git/github/${packageName}/${name}/${revision}`
+        else path = `git/github/${packageName}/${name}`
+        return path
+
+      case MAVEN_WEBSITE:
+        if (revision) path = `maven/mavencentral/${name}/${version}/${revision}`
+        else path = `maven/mavencentral/${name}/${version}`
+        return path
+
+      case NUGET_WEBSITE:
+        if (version) path = `nuget/nuget/-/${name}/${version}`
+        else path = `nuget/nuget/-/${name}`
+        return path
+
+      case PYPI_WEBSITE:
+        if (version) path = `pypi/pypi/-/${name}/${version}`
+        else path = `pypi/pypi/-/${name}`
+        return path
+
+      case RUBYGEM_WEBSITE:
+        if (revision) path = `gem/rubygems/-/${name}/${revision}`
+        else path = `gem/rubygems/-/${name}`
+        return path
+
+      default:
+        console.error('ATTENTION: Source provider not available!')
+        return null
```

Comment:
We should return an error, otherwise the user doesn't know what's happening

Reply:
Did it